lib/collection/variable-scope.js

```diff
@@ -177,8 +177,8 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
             i,
             ii;
 
-        // if a variable does not exist in local scope, we search all the layers and return the first occurence.
-        if (!variable && this._layers) {
+        // if a variable does not exist in local scope, we search all the layers and return the first occurrence.
+        if (!(variable || _.isEmpty(this._layers))) {
```

Comment:
We are creating this._layers on demand. Hence we had no chance of an empty array. Did we?